core/lib/dal/migrations/20231026192041_applied-snapshot-status.up.sql

```diff
@@ -0,0 +1,9 @@
+-- Add up migration script here
+
+CREATE TABLE applied_snapshot_status
+(
+    l1_batch_number         BIGINT NOT NULL PRIMARY KEY,
+    miniblock_number        BIGINT NOT NULL,
+    is_finished             BOOL NOT NULL,
+    last_finished_chunk_id   INT
```

Comment:
I wonder if we really need two tables `snapshots` and `applied_snapshot_status`, this way we would:
* enforce consistency: currently nothing prevents you from having a `applied_snapshot_status` with no matching `snapshots` instance. You could have NOT NULL Foregn key for that, but I don't see a reason for split.

Reply:
applied_snapshot_status is a table that is used only on EN side, where it keeps progress of applying the snapshot to EN instance. Is exists only to allow recovery process resilient i.e. that it could be resumed if it's interrupted at any point (though that part is very much still WIP). 
snapshots table keeps track of created snapshots and is used in snapshots API.
I decided to have different tables as applying snapshot needs completely different set of columns than the table used for snapshot management from the creator point of view.

Reply:
ah I understand better now - `snapshots` is when you are creating snapshots, while `applied_...` is for when you recover from snapshots. So there is no real 1-1 relationship